security/pkg/k8s/chiron/utils.go

```diff
@@ -253,6 +253,7 @@ func readSignedCsr(client clientset.Interface, csr string, watchTimeout time.Dur
 	if err != nil {
 		return nil, fmt.Errorf("failed to watch CSR %v", csr)
 	}
+	defer watcher.Stop()
```

Comment:
This change looks ~right for the current implementation, but we shouldn't be using raw watches in this code. It should be informer-based, preferably using krt

Reply:
Its because its for a single item + ephemeral. Also its copy+pasted from k8s code originally I believe (not sure if we can dedupe it now -- at the time I recall having some reasons we copied it)